ttnn/cpp/ttnn/operations/moreh/moreh_pow/device/kernels/moreh_pow_kernel.cpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "debug/dprint.h"
+#include "ttnn/cpp/ttnn/deprecated/tt_dnn/kernels/compute/moreh_common.hpp"
+
+namespace NAMESPACE {
+void MAIN {
+    int i{0};
+    const auto num_rows_per_core = get_arg_val<uint32_t>(i++);
+    const auto Wt = get_arg_val<uint32_t>(i++);
+    const auto origin_w = get_arg_val<uint32_t>(i++);
+    const auto p = get_arg_val<uint32_t>(i++);
+    const bool p_is_negative = get_arg_val<uint32_t>(i++) == 1;
+
+    std::uint8_t input_id{tt::CB::c_in0};
+    const auto cb_x = input_id++;        // input
+    const auto cb_one = input_id++;      // one
+    const auto cb_decimal = input_id++;  // decimal
+    const auto cb_mask_w = input_id++;   // mask_w
+
+    std::uint8_t output_id{tt::CB::c_out0};
+    const auto cb_y = output_id++;  // output
+
+    std::uint8_t intermed_id{tt::CB::c_intermed0};
+    const auto cb_tmp0 = intermed_id++;
+    const auto cb_tmp1 = intermed_id++;
+    const auto cb_tmp2 = intermed_id++;
+    const auto cb_tmp3 = intermed_id++;
+
+    const auto cb_xabs = cb_tmp0;      // |x|
+    const auto cb_xpow = cb_tmp1;      // |x|^p
+    const auto cb_logx = cb_tmp2;      // log(|x|)
+    const auto cb_exp_lxmd = cb_tmp3;  // exp(log(|x|) * decimal)
+
+    constexpr uint32_t onetile = 1;
+    constexpr uint32_t dst0 = 0;
+    constexpr uint32_t dst1 = 1;
+
+    binary_op_init_common(tt::CB::c_in0, tt::CB::c_in0);
+
+    cb_wait_front(cb_one, onetile);      // comes from the reader
+    cb_wait_front(cb_decimal, onetile);  // comes from the reader
+
+    constexpr uint32_t TILE_W = 32;
+    const bool do_mask_w = (origin_w % TILE_W) != 0;
+    const auto mask_w = do_mask_w ? (origin_w % TILE_W) : TILE_W;
+
+    if (do_mask_w) {
+        cb_wait_front(cb_mask_w, onetile);  // comes from the reader
+    }
+    for (uint32_t row_idx = 0; row_idx < num_rows_per_core; ++row_idx) {
+        for (uint32_t col_idx = 0; col_idx < Wt; ++col_idx) {
+            // |x|
+            tile_regs_acquire();
+            cb_wait_front(cb_x, onetile);  // comes from the reader
+            cb_reserve_back(cb_xabs, onetile);
+
+            copy_tile_init_with_dt(cb_x);
+            copy_tile(cb_x, 0, dst0);
+
+            if (do_mask_w && (col_idx == Wt - 1)) {
+                copy_tile_init_with_dt(cb_mask_w);
+                copy_tile(cb_mask_w, 0, dst1);
+
+                mask_tile_init();
+                mask_tile(dst0, dst1);
+            }
+
+            abs_tile_init();
+            abs_tile(dst0);
+            tile_regs_commit();
+
+            tile_regs_wait();
+            pack_tile_with_dt(dst0, cb_xabs);
+            tile_regs_release();
+
+            cb_pop_front(cb_x, onetile);
+            cb_push_back(cb_xabs, onetile);
+
+            power_tile_to_cb(cb_xabs, cb_xpow, cb_logx, cb_decimal, cb_exp_lxmd, cb_y, p, p_is_negative);
+        }
+    }
```

Comment:
The name "moreh_pow" can be confusing to people.
This is because what "moreh_pow" actually does is abs + pow.